akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/ActorContextAskSpec.scala

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (C) 2009-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.actor.typed.scaladsl
+
+import akka.actor.typed.{ ActorRef, Props, TypedAkkaSpec }
+import akka.pattern.AskTimeoutException
+import akka.testkit.typed.TestKit
+import akka.testkit.typed.scaladsl.TestProbe
+import akka.util.Timeout
+import com.typesafe.config.ConfigFactory
+
+import scala.concurrent.duration._
+import scala.reflect.ClassTag
+import scala.util.{ Failure, Success }
+
+object ActorContextAskSpec {
+  val config = ConfigFactory.parseString(
+    """
+      ping-pong-dispatcher {
+        executor = thread-pool-executor
+        type = PinnedDispatcher
+      }
+      snitch-dispatcher {
+        executor = thread-pool-executor
+        type = PinnedDispatcher
+      }
+    """)
+}
+
+class ActorContextAskSpec extends TestKit(ActorContextAskSpec.config) with TypedAkkaSpec {
+
+  "The Scala DSL ActorContext" must {
+
+    // FIXME we should have a test over remoting as well in typed-cluster
+
+    "provide a safe ask" in {
+      case class Ping(sender: ActorRef[Pong])
+      case class Pong(selfName: String, threadName: String)
+
+      val pingPong = spawn(Actor.immutable[Ping] { (ctx, msg) ⇒
+        msg.sender ! Pong(ctx.self.path.name, Thread.currentThread().getName)
+        Actor.same
+      }, "ping-pong", Props.empty.withDispatcherFromConfig("ping-pong-dispatcher"))
+
+      val probe = TestProbe[AnyRef]()
+
+      val snitch = Actor.deferred[Pong] { (ctx) ⇒
+
+        // Timeout comes from TypedAkkaSpec
+
+        ctx.ask(pingPong, Ping) {
+          case Success(pong) ⇒ Pong(ctx.self.path.name + "1", Thread.currentThread().getName)
```

Comment:
I guess it would make sense to also test the behavior when the result type isn't accepted by the asking actor. With this test `Pong` would be accepted anyway.

Reply:
Good idea (it will throw match error in the actor receive).